### PR TITLE
DT-1027 ✨ Support Recall and Indeterminate Sentence indicators

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/translator.kt [hunk 1]
```diff
@@ -46,6 +45,9 @@ fun <P:Prisoner> translate(prisoner : P, ob: OffenderBooking): P {
 
   // get the most serious offence for this booking
   prisoner.mostSeriousOffence = ob.offenceHistory?.firstOrNull{ off -> off.mostSerious && off.bookingId == ob.bookingId }?.offenceDescription
+  prisoner.recall = ob.recall
+  prisoner.legalStatus = ob.legalStatus
+  prisoner.indeterminateSentence = ob.sentenceTerms?.any{ st -> true == st.lifeSentence && st.bookingId == ob.bookingId }
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/translator.kt:50]: this reads a bit odd with the double equals on `true`, had to double check it wasn't javascript!
Would have thought that
```kotlin
  prisoner.indeterminateSentence = ob.sentenceTerms?.any{ it.lifeSentence && it.bookingId == ob.bookingId }
```
would be better

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/SentenceTerm.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.services.dto
+
+import java.time.LocalDate
+
+data class SentenceTerm (
+  var bookingId: Long? = null,
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/SentenceTerm.kt:6]: why the `var` here?

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/SentenceTerm.kt:6]: and are all the fields nullable?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/translator.kt (hunk 1 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/model/translator.kt:50]: yes not sure why I did that!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/SentenceTerm.kt (hunk 2 above)
Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/SentenceTerm.kt]: No - again corrected

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/OffenceHistoryDetail.kt [hunk 3]
```diff
@@ -3,11 +3,11 @@ package uk.gov.justice.digital.hmpps.prisonersearch.services.dto
 import java.time.LocalDate
 
 data class OffenceHistoryDetail (
-  val bookingId: Long?,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/OffenceHistoryDetail.kt:6]: Removed some null fields

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PersonalCareNeed.kt [hunk 4]
```diff
@@ -3,8 +3,8 @@ package uk.gov.justice.digital.hmpps.prisonersearch.services.dto
 import java.time.LocalDate
 
 data class PersonalCareNeed (
-  val problemType: String?,
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/dto/PersonalCareNeed.kt:6]: Removing nullable

**Pull request merged**